source/slang/slang-ir.h

```diff
@@ -260,6 +361,40 @@ struct IRInst
     template<typename T>
     T* findDecoration();
 
+        /// Get all the attributes attached to this instruction.
+    IROperandListBase getAllAttrs();
+
+        /// Find the first attribute of type `T` attached to this instruction.
+    template<typename T>
+    T* findAttr()
+    {
+        for( auto attr : getAllAttrs() )
+        {
+            if(auto found = as<T>(attr))
+                return found;
+        }
+        return nullptr;
+    }
+
+        /// Find all attributes of type `T` attached to this instruction.
+        ///
+        /// This operation assumes that attributes are grouped by type,
+        /// so that all the attributes of type `T` are contiguous.
+        ///
+    template<typename T>
+    IROperandList<T> findAttrs()
+    {
+        auto allAttrs = getAllAttrs();
```

Comment:
Good point. We have an IR validation pass and it should probably be enforcing this (I think it already enforces the rule that decorations come before regular instructions in any list of child instructions). The one gotcha is that we don't currently enable IR validation by default, in order to avoid any runtime cost.